### PR TITLE
[Durable Objects - Workers] Add Hibernation API to changelog; add note

### PR DESCRIPTION
**Pull request by elithrar**
This PR:

* Adds the new Hibernation API to the changelog
* Clarifies "why" a user should consider the Hibernation API (i.e. can reduce billable duration charges)
* Some formatting fixes for readability.

### CODE REVIEW

File: content/workers/runtime-apis/durable-objects.md [hunk 1]
```diff
@@ -359,9 +359,17 @@ The method takes a [`Request`](/workers/runtime-apis/request/) as the parameter
 
 If the method fails with an uncaught exception, the exception will be thrown into the calling Worker that made the `fetch()` request.
 
-### WebSockets Hibernation API (beta)
+### {{<beta>}} WebSockets Hibernation API {{</beta>}}
 
-Durable Objects WebSockets support includes Cloudflare-specific extensions to the standard WebSocket interface, related methods on the `state` object, and handler methods that a Durable Object can implement for processing WebSocket events. These APIs allow a Durable Object that is not currently running an event handler to be removed from memory while keeping its WebSockets connected ("hibernation").
+Durable Objects WebSockets support includes Cloudflare-specific extensions to the standard WebSocket interface, related methods on the `state` object, and handler methods that a Durable Object can implement for processing WebSocket events.
+
+The Hibernation APIs allow a Durable Object that is not currently running an event handler, such as handling a WebSocket message, HTTP request, or [alarm](/workers/learning/using-durable-objects/#alarms-in-durable-objects), to be removed from memory while keeping its WebSockets connected ("hibernation").
```
Comment (cacheburner) [inline content/workers/runtime-apis/durable-objects.md]: > The Hibernation APIs allow a Durable Object

The Hibernation **API allows** a Durable Object

File: content/workers/runtime-apis/durable-objects.md [hunk 2]
```diff
@@ -381,7 +390,8 @@ If an event occurs for a hibernated Durable Object's corresponding handler metho
...
+  - Adds a WebSocket to the set attached to this object. `ws.accept()` must NOT have been called separately. Once called, any incoming messages will be delivered by calling the Durable Object's `webSocketMessage()` handler, and `webSocketClose()` will be invoked upon disconnect. After calling this, the WebSocket is accepted, so its `send()` and `close()` methods can be used to send messages, but its `addEventListener()` method won't ever receive any events as they'll be delivered to the Durable Object instead. `tags` are optional string tags which can be used to look up the WebSocket with `getWebSockets()`. Each tag is limited to 256 characters, and each WebSocket is limited to 10 tags associated with it.
+  - This API has a maximum of 32,768 WebSockets connected per Durable Object instance, but the CPU and memory usage of a given workload may further limit the practical number of simultaneous connections.
```
Comment (cacheburner) [inline content/workers/runtime-apis/durable-objects.md:394]: > This API has a maximum of 32,768 WebSockets connected per Durable Object instance

This API **permits** a maximum of 32,768 **WebSocket connections** per Durable Object instance

File: content/workers/platform/changelog.md [hunk 3]
```diff
@@ -8,6 +8,10 @@ rss: file
 
 # Changelog
 
+## 2023-05-26
+
+- A new [Hibernation API](/workers/runtime-apis/durable-objects/#websockets-hibernation-api) (beta) has been added to [Durable Objects](/workers/learning/using-durable-objects/). The Hibernation API allows a Durable Object that is not currently running an event handler (for example: processing a WebSocket message or alarm) to be removed from memory while keeping its WebSockets connected (“hibernation”). A Durable Object that hibernates will not incur billable Duration (GB-sec) charges.
```
Comment (Maddy-Cloudflare) [inline content/workers/platform/changelog.md]:
```suggestion
- A new [Hibernation API](/workers/runtime-apis/durable-objects/#websockets-hibernation-api) (beta) has been added to [Durable Objects](/workers/learning/using-durable-objects/). The Hibernation API allows a Durable Object that is not currently running an event handler (for example, processing a WebSocket message or alarm) to be removed from memory while keeping its WebSockets connected (“hibernation”). A Durable Object that hibernates will not incur billable Duration (GB-sec) charges. 
```

File: content/workers/runtime-apis/durable-objects.md [hunk 4]
```diff
@@ -381,7 +390,8 @@ If an event occurs for a hibernated Durable Object's corresponding handler metho
...
+  - Adds a WebSocket to the set attached to this object. `ws.accept()` must NOT have been called separately. Once called, any incoming messages will be delivered by calling the Durable Object's `webSocketMessage()` handler, and `webSocketClose()` will be invoked upon disconnect. After calling this, the WebSocket is accepted, so its `send()` and `close()` methods can be used to send messages, but its `addEventListener()` method won't ever receive any events as they'll be delivered to the Durable Object instead. `tags` are optional string tags which can be used to look up the WebSocket with `getWebSockets()`. Each tag is limited to 256 characters, and each WebSocket is limited to 10 tags associated with it.
```
Comment (Maddy-Cloudflare) [inline content/workers/runtime-apis/durable-objects.md]:
```suggestion
  - Adds a WebSocket to the set attached to this object. `ws.accept()` must NOT have been called separately. Once called, any incoming messages will be delivered by calling the Durable Object's `webSocketMessage()` handler, and `webSocketClose()` will be invoked upon disconnect. After calling this, the WebSocket is accepted, so its `send()` and `close()` methods can be used to send messages. Its `addEventListener()` method won't ever receive any events as they'll be delivered to the Durable Object instead. `tags` are optional string tags which can be used to look up the WebSocket with `getWebSockets()`. Each tag is limited to 256 characters, and each WebSocket is limited to 10 tags associated with it.
```

**Review by Maddy-Cloudflare — APPROVED** [review]

File: content/workers/runtime-apis/durable-objects.md (hunk 1 above)
Comment (elithrar) [inline content/workers/runtime-apis/durable-objects.md]:
```suggestion
The Hibernation API allows a Durable Object that is not currently running an event handler, such as handling a WebSocket message, HTTP request, or [alarm](/workers/learning/using-durable-objects/#alarms-in-durable-objects), to be removed from memory while keeping its WebSockets connected ("hibernation").
```

File: content/workers/runtime-apis/durable-objects.md [hunk 5]
```diff
@@ -359,9 +359,17 @@ The method takes a [`Request`](/workers/runtime-apis/request/) as the parameter
 
 If the method fails with an uncaught exception, the exception will be thrown into the calling Worker that made the `fetch()` request.
 
-### WebSockets Hibernation API (beta)
+### {{<beta>}} WebSockets Hibernation API {{</beta>}}
```
Comment (jclee) [inline content/workers/runtime-apis/durable-objects.md:362]: Curious... It seems that this "beta" tag causes the section anchor to render as:

https://developers.cloudflare.com/workers/runtime-apis/durable-objects/#hahahugoshortcode-s2-hbhb

(can be seen by clicking on the "#" that appears to the left of the section tag when hovering over it.)

Comment (elithrar) [reply content/workers/runtime-apis/durable-objects.md:362]: Good catch. Wanted to get the `-beta` part out of the anchor tag so we don’t break it in future.

Let me set an explicit tag.

Comment (jclee) [reply content/workers/runtime-apis/durable-objects.md:362]: Looks like there are a few other places in the docs already using the "-beta" anchor.  Added PR #9302

**Pull request merged**